### PR TITLE
[Pull Request] Detaching cost calculation into a function and introducing new filters

### PR DESCRIPTION
**Pull request by iomatix**
```
Detached new cost as separate function 
the <int> getAttributeUpCost(key).

Added new filters at AttributeManager for {cost} (AttributeUpCost) and {invested} (InvestedAttribute)
```

### CODE REVIEW

**Comment by iomatix** [comment]
Tomorrow I'm gonna check some functions which are recalcluating spent aka invested attribute points at PlayerData.java
regarding this function:
```
public int getInvestedAttribute(String key) { return attributes.getOrDefault(key.toLowerCase(), 0); }
```
It may conflict with these new methods

**Comment by iomatix** [comment]
Yeah we'll need to check if distinguish spentAttributes is necessary 

but there's `attribPoints` and `attributes` but calculating 

_(line 90 at PlayerData.java)_ It regards this PlayerData hashmap: `HashMap<String, Integer> attributes`

_(line 403 at PlayerData.java)_ Check if the distinguish is necessary

Maybe only small fixes are necessary at _(line 369)_
```
    public int getInvestedAttribute(String key) {
        return attributes.getOrDefault(key.toLowerCase(), 0);
    }
```

**To the:**
```
    public int getInvestedAttribute(String key) {
        return attribPoints.getOrDefault(key.toLowerCase(), 0);
    }
```

**Comment by iomatix** [comment]
Almost done here for next pull request with PlayerData fixes

File: src/main/java/com/sucy/skill/api/player/PlayerData.java [hunk 1]
```diff
@@ -314,9 +315,21 @@ public HashMap<String, Integer> getInvestedAttributes() {
...
+     * @return attribute upgrade stages
+     */
+    public HashMap<String, Integer> getInvestedAttributesStages() {
+        return new HashMap<>(attrUpStages);
+    }
+
     /**
      * Gets the number of attribute points the player has
-     * between invested and bonus sources.
+     * from invested and bonus sources.
```
Comment (iomatix) [inline src/main/java/com/sucy/skill/api/player/PlayerData.java:332]: Yeah, I guess there was a typo cuz there is a sum so => 
it's (_from_) total = invested + bonus, not the substraction (_between_) invested - bonus

Comment (iomatix) [reply src/main/java/com/sucy/skill/api/player/PlayerData.java:332]: Shall we keep from for clarity maybe?

Comment (Travja) [reply src/main/java/com/sucy/skill/api/player/PlayerData.java:332]: Yeah, no reason why not.

**Comment by iomatix** [comment]
Changelog:
Attribute stage, cost_base and cost_modifier were introduced.
cost_base and cost_modifier are available at attributes.yml file.
cost_base: 1 - cost_base is a base cost of one upgrade of the attr stage. in this e.g. initial costs is 1
cost_modifier: 0.33334 - cost_modifier is a modifier to the base cost for each stage of the upgrade. in this e.g. costs will increase every 3

so:
# cost_base - initial cost of an attr upgrade
# cost_modifier - incremental value increasing cost_base for each of the attr upgrade

Filters for attribute manager has been updated to according changes.
# New placeholders for icon-lores:
# {amount} - the stage value for invested points.
# {total} - the stage value for invested points and class modifiers.
# {max} - max stage value player can invest the attr points.
# {cost} - cost in attr points for an upgrade to the next stage.
# {invested} - total spent attr points for the attr.
# {ap} - current value of attr points.

**Comment by Travja** [comment]
I think this is looking good, though could you revert changes in the `.github` folder and resolve merge conflicts for me?

**Comment by iomatix** [comment]
> I think this is looking good, though could you revert changes in the `.github` folder and pom.xml and resolve merge conflicts for me?

Rolled back them 😊~

File: .github/workflows/maven-publish.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#apache-maven-with-a-settings-path
+
+name: Maven Package
```
Comment (Travja) [inline .github/workflows/maven-publish.yml:4]: This workflow in particular should be removed for merge

**Review by Travja — CHANGES_REQUESTED** [review]

File: .github/workflows/maven-publish.yml (hunk 2 above)
Comment (iomatix) [reply .github/workflows/maven-publish.yml:4]: Okaay

**Review by Travja — APPROVED** [review]

**Pull request merged**